### PR TITLE
[react] Add `globalCss` processor

### LINKED ISSUE #9: [core] Support the Global Styles API
### Summary

Provide a similar API to Emotion/Styled-components `GlobalStyles`. 

### Examples

https://mui.com/material-ui/customization/how-to-customize/#4-global-css-override

### Motivation

Feature parity for migrating from Emotion/Styled-components.

**Search keywords**: pigment-css global styles

### PR DESCRIPTION
**Pull request by siriwatknp**
closes #9 

Material UI components need this.

- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/mui/pigment-css/blob/HEAD/CONTRIBUTING.md#sending-a-pull-request).

### CODE REVIEW

File: packages/pigment-css-react/tests/globalCss/globalCss.test.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import path from 'node:path';
+import { runTransformation, expect } from '../testUtils';
+
+describe('Pigment CSS - globalCss', () => {
```
[the remark this answers is not in the document]
Comment (siriwatknp) [reply packages/pigment-css-react/tests/globalCss/globalCss.test.ts:4]: Added.

**Comment by brijeshb42** [comment]
Will the README for the API be updated separately or should it be done in this PR itself ?

**Comment by siriwatknp** [comment]
> Will the README for the API be updated separately or should it be done in this PR itself ?

Good point, I will update this PR.

**Comment by siriwatknp** [comment]
@brijeshb42 Please check https://github.com/mui/pigment-css/pull/31/commits/1f968f0b87bf46e81b192b3a4cc24acf54017cdf

File: README.md [hunk 2]
```diff
@@ -467,6 +467,25 @@ function App() {
...
+
+globalCss`
+  body {
+    margin: 0;
+    padding: 0;
+  }
+`;
+```
+
+The `globalCss` function should to be called at the top level of your javascript file, usually from the index of the application.
```
Comment (brijeshb42) [inline README.md:485]:
```suggestion
The `globalCss` function should to be called at the top level of your javascript file, usually from the entry point of the application.
```

**Comment by siriwatknp** [comment]
@mnajdova @brijeshb42 Can I get an approval for this? It's blocking https://github.com/mui/material-ui/pull/42431

**Review by brijeshb42 — APPROVED** [review]

**Pull request merged**